cmake/RetrieveGoogleTest.cmake

```diff
@@ -0,0 +1,50 @@
+##
+## CMake script to download GoogleTest and
+## add to the project runtime.
+##
+## See http://kaizou.org/2014/11/gtest-cmake/
+
+# We need thread support
+find_package(Threads REQUIRED)
+
+# Enable ExternalProject CMake module
+include(ExternalProject)
+
+# Download and install GoogleTest
+ExternalProject_Add(
+        gtest
+        URL https://github.com/google/googletest/archive/master.zip
+        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
+        CMAKE_ARGS
+          -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
```

Comment:
`CMAKE_ARGS` needed during build of **gtest** to build under `clang++-4.0` properly.